### PR TITLE
INFINITY-2601, INFINITY-2496 CLI: faster, better error message, support service_name setting

### PR DESCRIPTION
**Pull request by nickbp**
- Fix the inaccurate messaging when a 404 is encountered.
- Attempt to read the toml file directly to save time, but fall back to running 'dcos' directly as before if that fails for some reason
- Fix the undocumented '<modname>.service_name' setting, and document it in relevant error messages

Demo!:

1. Speed:

BEFORE:
```
$ time dcos kafka pod list
[
  "kafka-0",
  "kafka-1",
  "kafka-2"
]

real	0m2.297s
user	0m2.020s
sys	0m0.240s
```
AFTER:
```
$ time dcos beta-kafka pod list
[
  "kafka-0",
  "kafka-1",
  "kafka-2"
]

real	0m0.522s
user	0m0.360s
sys	0m0.072s
```
(or faster depending on your network)

2. 404 fixed using `<pkg>.service_name`:

```
$ dcos beta-kafka pod list
Got 404 Not Found response from http://nick-plsx-elasticl-a9tyx0cj04fx-1638172899.us-west-2.elb.amazonaws.com/service/beta-kafka/v1/pod:
- The service scheduler may have been unable to find an item that was specified in your request.
- The DC/OS cluster may have been unable to find a service named "beta-kafka". Specify a service name with '--name=<name>', or with 'dcos config set beta-kafka.service_name <name>'.

$ dcos config set beta-kafka.service_name kafka
[beta-kafka.service_name]: set to 'kafka'

$ dcos beta-kafka pod list
[
  "kafka-0",
  "kafka-1",
  "kafka-2"
]
```

### CODE REVIEW

File: cli/client/dcoscli_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+func TestDcosCliTestSuite(t *testing.T) {
+	suite.Run(t, new(DcosCliTestSuite))
+}
+
+func (suite *DcosCliTestSuite) TestGetCLIValues() {
+	cachedConfig = nil
+
+	config, err := cliDiskConfig("testdata/config_missing")
+	assert.Equal(suite.T(), "open testdata/config_missing/.dcos/clusters: no such file or directory", err.Error())
```
Comment (benclarkwood) [inline cli/client/dcoscli_test.go]: I'd just assert that there is an error.

I'm not a big fan of having to update text in multiple places.

Comment (benclarkwood) [reply cli/client/dcoscli_test.go]: Or just assert containment or some such.

File: cli/client/dcoscli.go [hunk 2]
```diff
@@ -49,10 +59,102 @@ func RequiredCLIConfigValue(name string, description string, errorInstruction st
...
+			continue // not a directory
+		}
+		if _, err := os.Stat(path.Join(clustersPath, cluster.Name(), "attached")); os.IsNotExist(err) {
+			continue // not the attached/active cluster
+		}
+		configPath = path.Join(clustersPath, cluster.Name(), "dcos.toml")
+		break
+	}
+
+	if configPath == "" {
```
Comment (benclarkwood) [inline cli/client/dcoscli.go]: I can't tell if any of the tests actually exercise this fall back?

File: cli/client/dcoscli.go [hunk 3]
```diff
@@ -49,10 +58,112 @@ func RequiredCLIConfigValue(name string, description string, errorInstruction st
...
+				continue // not the attached/active cluster
+			}
+			configPath = path.Join(clustersPath, cluster.Name(), "dcos.toml")
+			break
+		}
+	}
+
+	// Fall back to old-style single config:
+	if configPath == "" {
+		configPath = path.Join(configDir, "dcos.toml")
```
Comment (benclarkwood) [inline cli/client/dcoscli.go]: I can't tell, is this exercised in any of the tests?

File: cli/client/dcoscli_test.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+func TestDcosCliTestSuite(t *testing.T) {
+	suite.Run(t, new(DcosCliTestSuite))
+}
+
+func (suite *DcosCliTestSuite) TestGetCLIValues() {
+	cachedConfig = nil
+
+	config, err := cliDiskConfig("testdata/cliconfig_missing")
+	assert.Equal(suite.T(), "open testdata/cliconfig_missing/dcos.toml: no such file or directory", err.Error())
```
Comment (benclarkwood) [inline cli/client/dcoscli_test.go]: My preference would be to assert an error exists and check containment of some specific string if really necessary. I dislike updating text in multiple places.

Comment (benclarkwood) [reply cli/client/dcoscli_test.go]: Similar thoughts for other equals assertions on errors.

File: cli/client/dcoscli.go (hunk 3 above)
Comment (nickbp) [reply cli/client/dcoscli.go]: Added a test case for a directory which only has an old config

File: cli/client/dcoscli_test.go (hunk 4 above)
Comment (nickbp) [reply cli/client/dcoscli_test.go]: Switched to Contains when Equal was being especially verbose

**Comment by nickbp** [comment]
TODO: Config dir lookup doesn't work when manually tested on windows. This isn't the end of the world as it's just falling back to the prior behavior. I'll see if I can find the correct path on windows, and update the config retrieval to try there as well.

```
>dcos beta-cassandra -v pod list
Unable to resolve CLI config directory (DCOS_DIR or HOME/.dcos), falling back to running CLI command.
Running DC/OS CLI command: dcos config show core.dcos_url
Unable to resolve CLI config directory (DCOS_DIR or HOME/.dcos), falling back to running CLI command.
Running DC/OS CLI command: dcos config show core.dcos_acs_token
HTTP Query: GET http://nick-ym2i-elasticl-13syk38h0atvx-16587567.us-west-2.elb.amazonaws.com/service/beta-cassandra/v1/pod
```

**Pull request merged**